library/src/main/java/com/novoda/downloadmanager/RoomDownloadsPersistence.java

```diff
@@ -139,4 +139,46 @@ public boolean update(DownloadBatchId downloadBatchId, boolean notificationSeen)
         database.roomBatchDao().update(roomBatch);
         return true;
     }
+
+    @Override
+    public void persistCompletedBatch(CompletedDownloadBatch completedDownloadBatch) {
+        DownloadBatchId downloadBatchId = completedDownloadBatch.downloadBatchId();
+        DownloadBatchTitle downloadBatchTitle = completedDownloadBatch.downloadBatchTitle();
+        DownloadBatchStatus.Status downloadBatchStatus = DownloadBatchStatus.Status.DOWNLOADED;
+        long downloadedDateTimeInMillis = completedDownloadBatch.downloadedDateTimeInMillis();
+
+        DownloadsBatchPersisted persistedBatch = new LiteDownloadsBatchPersisted(
+                downloadBatchTitle,
+                downloadBatchId,
+                downloadBatchStatus,
+                downloadedDateTimeInMillis,
+                true
+        );
+        persistBatch(persistedBatch);
```

Comment:
Do we need to check for duplicated batch ids? Maybe it's worth failing fast, maybe exposing an explicit Exception in the method signature, so that clients are aware of avoiding duplicates